documentation/ranking.html

```diff
@@ -4,27 +4,91 @@
 ---
 
 <p>
-Vespa uses <a href="reference/ranking-expressions.html">ranking expressions</a>
-to rank documents matching a query, computing a rank score per document.
-A ranking expression is stored in a <a href="#rank-profile">rank profile</a>.
-An application can have multiple rank profiles -
-this can be used for implementing different use cases, or bucket testing ranking variations.
-</p><p>
-Example: Rank the freshest document highest, using the value in an attribute named <em>timestamp</em>:
+Vespa defines Big Data Serving as:
+<blockquote style="background: #1a7db6; color: #ffffff"><strong>
+Selection, organization and machine-learned model inference,
+<ul>
+    <li>over many, constantly changing data items (thousands to billions),</li>
+    <li>with low latency (~100 ms) and high load (thousands of queries/second)</li>
+</ul>
+</strong></blockquote>
+<em>Ranking</em> enables organization and ML inference,
+and <em>multi-phased ranking</em> addresses latency and load:
 <pre>
 search myapp {
-    &hellip;
+
     rank-profile my-rank-profile {
         first-phase {
-            expression: freshness(timestamp)
+            expression: attribute(quality) * freshness(timestamp)
+        }
+        second-phase {
+            expression: sum(onnx("my-model.onnx", "add"))
+        }
+    }
+}
+</pre>
+Applications use the <a href="search-api.html">query API</a> to <em>select</em> the documents to evaluate
+using a <a href="query-language.html">query language</a>,
+and choose a <a href="#rank-profile">rank profile</a> for the ranking.
+Rank profiles have <a href="#two-phase-ranking">one or two phases</a>:
+<ul>
+    <li>Phase one should use a computationally inexpensive function to rank candidates</li>
+    <li>Phase two is run on a small candidate set</li>
+</ul>
+Hence, the query selection and first phase ranking reduces the size of the computation -
+then machine-learned models can be used on the second-phase ranking on
+<a href="reference/search-definitions-reference.html#rerank-count">rerank-count</a> documents per node.
+This makes the ranking scalable:
+<ul>
+    <li>Control the second phase candidate set size</li>
+    <li>Add content nodes to rank more documents / less documents per node</li>
+</ul>
+</p><p>
+Ranking is run in the <a href="overview.html">content cluster</a>.
+This means, <em>online inference</em> (a.k.a. <em>real time inference</em> or <em>dynamic inference</em>)
+is executed for the candidate document set.
+Ranking is running <a href="#ranking-expressions">ranking expressions</a> using
+<a href="rank-features">rank features</a> (values / computed values from queries, document and constants).
+</p><p>
+Note: Vespa also supports <a href="stateless-model-evaluation.html">stateless model evaluation</a> -
+making inferences without documents (i.e. query to model).
+</p>
+
+
+
+<h2 id="machine-learned-model-inference">Machine-Learned model inference</h2>
+<p>
+Vespa supports the following ML models:
```

Comment:
Not exclusive list, we can represent lots of model, logistic regression etc. Also mention PyTorch through ONNX.

Reply:
added todo